src/pal/pal_posix.h

```diff
@@ -196,6 +197,14 @@ namespace snmalloc
 
         if (r != MAP_FAILED)
           return;
+
+        /*
+         * We're falling back to zeroing the memory ourselves, which is not
+         * stellar, but correct.  However, mmap() has left errno nonzero in an
+         * effort to explain that MAP_FAILED.  Reset it here so subsequent
+         * checks don't see a stale value.
+         */
+        errno = 0;
```

Comment:
This is not actually POSIXly correct: the spec for `errno` says that it should not be modified by things that don't explicitly set it.  We should be getting `errno` before calling the thing that can set errno and resetting it later.  It would be quite nice to capture this in an RAII thing (capture errno in constructor, reset errno in destructor).